### DISCUSSION TITLE
About messages output

### DISCUSSION THREAD

**Discussion by MuuXB** [zauberzeug/nicegui#3566] [Q&A] [answered]
### Question

1. I have a function that receives messages in real-time. I want to display these messages in real-time through the NiceGUI interface. Do you have any good ideas for this? Furthermore, I want to implement scrollable output by limiting the total number of messages.
2. A button triggers a function, and there is an area on the interface that displays the function's result. Clicking different buttons will output the result to this area (replacing the previous result).

Looking forward to someone answering thanks

**Comment by python-and-novella** [comment, marked as answer]
Maybe `ui.log` is you wanted.
